compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal

```diff
@@ -0,0 +1,156 @@
+    }
+
+    remote function onError(kafka:Error 'error) returns error|() {
+    }
+}
+
+service kafka:Service on kafkaListener {
+    private final string var1 = "Kafka Service";
+    private final int var2 = 54;
+
+    remote function onConsumerRecord(@kafka:Payload PayloadConsumerRecord[] & readonly data, kafka:ConsumerRecord[] records, kafka:Caller caller) {
+    }
+
+    remote function onError(kafka:Error 'error) returns error|() {
+    }
+}
+
+service kafka:Service on kafkaListener {
+    private final string var1 = "Kafka Service";
+    private final int var2 = 54;
+
+    remote function onConsumerRecord(@kafka:Payload PayloadConsumerRecord[] & readonly data, kafka:ConsumerRecord[] & readonly records, kafka:Caller caller) {
+    }
+
+    remote function onError(kafka:Error 'error) returns error|() {
+    }
+}
+
+service on kafkaListener {
+    private final string var1 = "Kafka Service";
+    private final int var2 = 54;
+
+    remote function onConsumerRecord(@kafka:Payload PayloadConsumerRecord[] & readonly data, PersonConsumerRecord[] & readonly records, kafka:Caller caller) {
+    }
+
+    remote function onError(kafka:Error 'error) returns error|() {
+    }
+}
+
+service on kafkaListener {
+    private final string var1 = "Kafka Service";
+    private final int var2 = 54;
+
+    remote function onConsumerRecord(@kafka:Payload PayloadConsumerRecordWithTypeReference[] & readonly data, PersonConsumerRecord[] & readonly records, kafka:Caller caller) {
+    }
+
+    remote function onError(kafka:Error 'error) returns error|() {
+    }
+}
+
+service on kafkaListener {
+    private final string var1 = "Kafka Service";
+    private final int var2 = 54;
+
+    remote function onConsumerRecord(string[] & readonly data, PersonConsumerRecord[] & readonly records, kafka:Caller caller) {
+    }
+
+    remote function onError(kafka:Error 'error) returns error|() {
+    }
+}
+
+service on kafkaListener {
+    private final string var1 = "Kafka Service";
+    private final int var2 = 54;
+
+    remote function onConsumerRecord(string[] & readonly data, kafka:ConsumerRecord[] & readonly records, kafka:Caller caller) {
+    }
+
+    remote function onError(kafka:Error 'error) returns error|() {
+    }
+}
+
+public type PersonConsumerRecord record {|
+    *kafka:AnydataConsumerRecord;
+    Person value;
+|};
+
+public type PayloadConsumerRecord record {|
+    string key?;
+    string value;
+    int timestamp;
+    record {|
+        int offset;
+        record {|
+            string topic;
+            int partition;
+        |} partition;
+    |} offset;
+|};
+
+public type PayloadConsumerRecordWithTypeReference record {|
+    string key?;
+    string value;
+    int timestamp;
+    kafka:PartitionOffset offset;
+|};
+
+public type Person record {|
+    string name;
+    boolean isMarried;
+    int age;
+|};
```

Comment:
Also let's add scenario like the below.

```
remote function onConsumerRecord(@kafka:Payload kafka:MyRecord[] records)
```

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/508/commits/c75fc2d0126153e4eb6265ec75ee9a04af23a17f